### PR TITLE
 refactor!: use big endian and simplify transcript logic

### LINKED ISSUE #1797: PolynomialEval_evalDataGen_Test test failure
A contract test fails in the `commonprefix-patch` branch.

```
> git log
commit ffc6122af57ccec46a0d2d364a0db99d30735d5f (HEAD -> commonprefix-patch, origin/commonprefix-patch)
Author: Alex Xiong <alex.xiong.tech@gmail.com>
Date:   Fri Jul 26 22:32:08 2024 +0800

    chore: update hotshot to 0.1.60, most jf dep to 0.4.5 (#1788)

> just sol-test
Failing tests:
Encountered 1 failing test in contracts/test/PolynomialEval.t.sol:PolynomialEval_evalDataGen_Test
[FAIL. Reason: assertion failed: 0 != 62; counterexample: calldata=0x0c8cc1400000000000000000000000000000000000000000000000000000000000001e4600eeb2cb5981ed45649abebde081dcff16c8601de4347e7dd1628ba2daac43b7000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000ec000000000000000000000000000000000000000000000000000000000007ffff0000000000000000000000000000000000000000000000000000000000001fca0000000000000000000000000000000000000000000000000000000000001eb900000000000000000000000000000000000000000000000000000000000005cf00000000000000000000000000000000000000000000000000000000000002cf00000000000000000000000000000000000000000000000000000000b11a19e9000000000000000000000000000000000000000000000000000000000000202e0000000000000000000000000000000000000000000000000000000000000cf60000000000000000000000000000000000000000000000000000000000001edd00000000000000000000000000000000000000000000000000000000000005d000000000000000000000000000000000000000000000000000000000260a5b160000000000000000000000000000000000000000000000000000000000001edd0000000000000000000000000000000000000000000000191a59998b5d195cdd000000000000000000000000000000000000000000000000000000000000102f0000000000000000000000000000000000000000000000000000000000001edc0000000000000000000000000000000000000000000000000000000000001e38000000000000000000000000000000000000000000000000000000000000003e00000000000000000000000000000000000000000000000000000000000013f500000000000000000000000000000000000000000000000000000000000018d400000000000000000000000000000000000000000000000000000000000003ae00000000000000000000000000000000000000000000000000000000ba414fa730644b6c9c4a72169e4daa317d25f04512ae15c53b34e8f5acd8e155d0a6c100000000000000000000000000000000000000000000000000000000000000157a0000000000000000000000000000000000000000000000000000000000001bc10000000000000000000000000000000000000000000000000000000000001ee9000000000000000000000000000000000000000000000000000000000000142100000000000000000000000000000000000000000000000000000000000016610000000000000000000000000000000000000000000000000000000000000377000000000000000000000000000000000000000000000000000000000000025a000000000000000000000000000000000000000000000000000000000000162400000000000000000000000000000000000000000000000000000000000010290000000000000000000000000000000000000000000000000000000000001c5a00000000000000000000000000000000000000000000000000000000000012b30000000000000000000000000000000000000000000000000000000000001f25000000000000000000000000000000000000000000000000000000000000172f0000000000000000000000000000000000000000000000000000000000001f7000000000000000000000000000000000000000000000000000000000000018880000000000000000000000000000000000000000000000000000000000001e9000000000000000000000000000000000000000000000000000000000667f9d7000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000001f542260e724844bca5251829353968e4915305258418357473a5c1d597f613f6cbe00000000000000000000000000000000000000000000000000000000000018b100000000000000000000000000000000000000000000000000000000000009770000000000000000000000000000000000000000000000000000000000001aad00000000000000000000000000000000000000000000000000000000000019a10000000000000000000000000000000000000000000000000000000000001cf50000000000000000000000000000000000000000000000000000000000000e48426e3235343a20706f7720707265636f6d70696c65206661696c6564210000000000000000000000000000000000000000000000000000000000000000001ab000000000000000000000000000000000000000000000000000000000000001bc30644b6c9c4a72169e4daa317d25f04512ae15c53b34e8f5acd8e155d0a6c1022ee12bff4a2813286a8dc388cd754d9a3ef2490635eba50cb9c2e5e75080000000000000000000000000000000000000000000000000000000000000000016f70000000000000000000000000000000000000000000000000000000071aad10d000000000000000000000000000009709ecfa91a80626ff3989d68f67f5b1dd1000000000000000000000000000000000000000000000000000000000000134700000000000000000000000000000000000000000000000000000000000003200000000000109b8c8d4d0e881a5b9d985b1a59081cd8d85b185c88199a595b1800000000000000000000000000000000000000000000000000000000000000f90000000000000000000000000000000000000000000000000000000000000f1e0000000000000000000000000000000000000000000000000000000000001f6700000000000000000000000000000000000000000000000000000000000006850000000000000000000000000000000000000000000000000000000000001f1c00000000000000000000000000000000000000000000000000000000000014a100000000000000000000000000000000000000000000000000000000000016f5000000000000000000000000000000000000000000000000000000000000204900000000000000000000000000000000000000000000000000000000000009e30000000000000000000000000000000000000000000000000000000000001aae00000000000000000000000000000000000000000000000000000000000015bd0000000000000000000000000000000000000000000000000000000000000823000000000000000000000000000000000000000000000000000000005c445151000000000000000000000000000000000000000000000000000000000000157a000000000000000000000000000000000000000000000000000000000000012700000000000000000000000000000000000000000000000000000000fa7626d30000000000000000000000000000000000000000000000000000000000001fba000000000000000000000000000000000000000000000000000000000000171f000000000000000000000000000000000000000000000000000000000000213b00000000000000000000000000000000000000000000000000000000000016bf244cf010c43ca87237d8b00bf9dd50c4c01c7f086bd4e8c920e75251d96f0d2100000000000000000000000000000000000000000000000000000000000000210000000000000000000000000000000000000000000000000000000000001f1a0000000000000000000000000000000000000000000000000000000000001f520000000000000000000000000000000000000000000000000000000000000a0700000000000000000000000000000000000000000000000000000000000019fc00000000000000000000000000000000000000000000000000000000000017c1000000000000000000000000000000000000000000000000000000000000026a00000000000000000000000000000000000000000000000000000000000002bb000000000000000000000000000000000000000000000000000000000000018d00000000000000000000000000000000000000000000000000000000000000610000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000121f0000000000000000000000000000000000000000000000000000000000000baa3064486657634403844b0eac78ca882cfd284341fcb0615a15cfcd17b14d820200000000000000000000000000000000000000000000000000000000000017b9426e3235343a20706f7720707265636f6d70696c65206661696c656420ffffff00000000000000000000000000000000000000000000000000000000000018ea0000000000000000000000000000000000000000000000000000000000000cbe00000000000000000000000000000000000000000000000000000000000005a9000000000000000000000000000000000000000000000000000000005e5d6098000000000000000000000000000000000000000000000000000000000000006500000000000000000000000000000000000000000000000000000000e2ef09e40000000000000000000000000000000000000000000000000000000000001c120000000000000000000000000000000000000000000000000000000000000fc9000000000000000000000000000000000000000000000000000000000000013d000000000000000000000000000000000000000000000000000000000000191e0000000000000000000000000000000000000000000000000000000000001b2a00000000000000000000000000000000000000000000000000000000000000e30000000000000000000000000000000000000000000000000000000000001ec70000000000000000000000000000000000000000000000000000000000001dbf0000000000000000000000000000000000000000000000000000000085226c82000000000000000000000000000000000000000000000000000000000000179500000000000000000000000000000000000000000000000000000000000017ed000000000000000000000000000000000000000000000000000000000000013f00000000000000000000000000000000000000000000000000000000000017e20000000000000000000000000000000000000000000000000000000000000039000000000000000000000000000000000000000000000000000000002ade387f0000000000000000000000000000000000000000000000000000000000001a9d244cf010c43ca87237d8b00bf9dd50c4c01c7f086bd4e8c920e75251d96f0d2100000000000000000000000000000000000000000000000000000000000019af00000000000000000000000000000000000000000000000000000000fa7626d500000000000000000000000000000000000000000000000000000000000005110000000000000000000000000000000000000000000000000000000000000db80000000000000000000000000000000000000000000000000000000000000f1e00000000000000000000000000000000000000000000000000000000000016a300000000000000000000000000000000000000000000000000000000000018c1000000000000000000000000000000000000000000000000000000000000161c000000000000000000000000000000000000000000000000000000000000176500000000000000000000000000000000000000000000000000000000000017e0000000000000000000000000000000000000000000000000000000000000208a000000000000000000000000000000000000000000000000000000000000000d000000000000000000000000000000000000000000000000000000000000075500000000000000000000000000000000000000000000000000000000000000bb00000000000000000000000000000000000000000000000000000000000001080000000000000000000000000000000000000000000000000000000000001f7000000000000000000000000000000000000000000000000000000000000016da00000000000000000000000000000000000000000000000000000000000003ad000000000000000000000000000000000000000000000000000000000000189f00000000000000000000000000000000000000000000000000000000000003f10000000000000000000000000000000000000000000000000000000000001c930000000000000000000000000000000000000000000000000000000000002105000000000000000000000000000000000000000000000000000000000000161e0000000000000000000000000000000000000000000000000000000000000c6f0000000000000000000000000000000000000000000000000000000000001ff2000000000000000000000000000000000000000000000000000000000000005e0000000000000000000000000000000000000000000000000000000000020001000000000000000000000000000000000000000000000000000000001ed7831c0000000000000000000000000000000000000000000000000000000000000a8800000000000000000000000000000000000000000000000000000000000000bf000000000000000000000000000000000000000000000000000000000000010630644259cd94e7dd5045d7a27013b7fcd21c9e3b7fa75222e7bda49b729b04020000000000000000000000000000000000000000000000000000000000001f01000000000000000000000000000000000000000000000000000000000000212900000000000000000000000000000000000000000000000000000000000000e10000000000000000000000000000000000000000000000000000000000001956000000000000000000000000000000000000000000000000000000000000099200000000000000000000000000000000000000000000000000000000000007080000000000000000000000000000000000000000000000000000000000000362000000000000000000000000000000000000000000000000000000008c5e11f200000000000000000000000000000000000000000000000000000000000000f900000000000000000000000000000000000000000000000000000000000002700000000000000000000000000000000000000000000000000000000000002123000000000000000000000000000000000000000000000000000000000000004400000000000000000000000000000000000000000000000000000000000000dd00000000000000000000000000000000000000000000000000000000fa7626d400000000000000000000000000000000000000000000000000000000000017c000000000000000000000000000000000000000000000000000000000000017f40000000000000000000000000000000000000000000000000000000000001995000000000000000000000000000000000000000000000000000000000000157b0000000000000000000000000000000000000000000000000000000000000a880000000000000000000000000000000000000000000000000000000000000933000000000000000000000000000000000000000000000000000000000000186d0000000000000000000000000000000000000000000000000000000000001d56000000000000000000000000000000000000000000000000000000002631f2b10000000000000000000000000000000000000000000000000000000000000af6426e3235343a20706f7720707265636f6d70696c65206661696c6564210000000000000000000000000000000000000000000000000000000000000000000f931bf82deba7d74902c3708cc6e70e61f30512eca95655210e276e5858ce8f58e40000000000000000000000000000000000000000000000000000000000000d6206e402c0a314fb67a15cf806664ae1b722dbc0efe66e6c81d98f9924ca535322000000000000000000000000000000000000000000000000000000000000075530643640b9f82f90e83b698e5ea6179c7c05542e859533b48b9953a2f53608020000000000000000000000000000000000000000000000000000000000001f940000000000000000000000000000000000000000000000000000000000000b0900000000000000000000000000000000000000000000000000000000000003ad26125da10a0ed06327508aba06d1e303ac616632dbed349f53422da95333785700000000000000000000000000000000000000000000000000000000000000a2000000000000000000000000000000000000000000000000000000000000097500000000000000000000000000000000000000000000000000000000000001d00000000000000000000000000000000000000000000000000000000000001454000000000000000000000000000000000000000000000000000000003e5e3c22000000000000000000000000000000000000000000000000000000000000001d00000000000000000000000000000000000000000000000000000000000014db0000000000000000000000000000000000000000000000000000000000001c390000000000000000000000000000000000000000000000191a59998b5d195cdd000000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000000000000000000000000000000000016b30000000000000000000000000000000000000000000000000000000000001f25000000000000000000000000000000000000000000000000000000000000070800000000000000000000000000000000000000000000000000000000000019aa00000000000000000000000000000000000000000000000000000000000010f300000000000000000000000000000000000000000000000000000000000018d5000000000000000000000000000000000000000000000000000000000000070900000000000000000000000000000000000000000000000000000000000001bd00000000000000000000000000000000000000000000000000000000000000220000000000000000000000000000000000000000000000000000000000000ae9000000000000000000000000000000000000000000000000000000000000044600000000000000000000000000000000000000000000000000000000000000e6000000000000000000000000000000000000000000000000000000000000016a0000000000000000000000000000000000000000000000000000000000000270000000000000000000000000000000000000000000000000000000000000010f00000000000000000000000000000000000000000000000000000000000008540000000000000000000000000000000000000000000000000000000000000e88000000000000000000000000000000000000000000000000000000000000001d0000000000000000000000000000000000000000000000000000000000000ddc0000000000000000000000000000000000000000000000000000000000000da00000000000000000000000000000000000000000000000000000000000001b160000000000000000000000000000000000000000000000000000000000001ec600000000000000000000000000000000000000000000000000000000000016d9000000000000000000000000000000000000000000000000000000000000027000000000000000000000000000000000000000000000000000000000000020d500000000000000000000000000000000000000000000000000000000000002240000000000000000000000000000000000000032bb30b616b230ba3096b3b2b6000000000000000000000000000000000000000000000000000000000000009a000000000000000000000000000000000000000000000000000000000000004400000000000000000000000000000000000000000000000000000000000005130000000000000000000000000000000000000000000000000000000000000ca000000000000000000000000000000000000000000000000000000000000016c000000000000000000000000000000000000000000000000000000000000007f2000000000000000000000000000000000000000000000000000000000000087300000000000000000000000000000000000000000000000000000000916a17c72ee12bff4a2813286a8dc388cd754d9a3ef2490635eba50cb9c2e5e7508000000000000000000000000000000000000000000000000000000000000000000f1f00000000000000000000000000000000000000000000000000000000000000be0000000000000000000000000000000000000000000000000000000000001d93000000000000000000000000000000000000000000000000000000000000093f args=[7750, 421743594562400382753388642386256516545992082196004333756405989743524594615 [4.217e74], [524287 [5.242e5], 8138, 7865, 1487, 719, 2971277801 [2.971e9], 8238, 3318, 7901, 1488, 638212886 [6.382e8], 7901, 463067319384493612253 [4.63e20], 4143, 7900, 7736, 62, 5109, 6356, 942, 3124842407 [3.124e9], 21888221997584217086951279548962733484243966294447177135413498358668068307200 [2.188e76], 5498, 7105, 7913, 5153, 5729, 887, 602, 5668, 4137, 7258, 4787, 7973, 5935, 8048, 6280, 7824, 1719639408 [1.719e9], 1, 8020, 15549849457946371566896172786938980432421851627449396898353380550861104573630 [1.554e76], 6321, 2423, 6829, 6561, 7413, 3656, 30047347705315867906698902152804724170386939875701103062026714790719945441280 [3.004e76], 6832, 444, 21888221997584217086951279548962733484243966294447177135413498358668068307202 [2.188e76], 21204235282094297871551205565717985242031228012903033270457635305745314480128 [2.12e76], 5879, 1907020045 [1.907e9], 822333294739597949471917823180105001410001 [8.223e41], 4935, 800, 6831975885078433717065899738591413112724687618901888805111421720 [6.831e63], 249, 3870, 8039, 1669, 7964, 5281, 5877, 8265, 2531, 6830, 5565, 2083, 1547981137 [1.547e9], 5498, 295, 4202047187 [4.202e9], 8122, 5919, 8507, 5823, 16419199797069286018154632093538978307443519224016378057156716264399020428577 [1.641e76], 33, 7962, 8018, 2567, 6652, 6081, 618, 699, 397, 97, 6, 4639, 2986, 21888201123329158951656153352668191879939568188478319927128792530760328118786 [2.188e76], 6073, 30047347705315867906698902152804724170386939875701103062026714790719945441279 [3.004e76], 6378, 3262, 1449, 1583177880 [1.583e9], 101, 3807316452 [3.807e9], 7186, 4041, 317, 6430, 6954, 227, 7879, 7615, 2233625730 [2.233e9], 6037, 6125, 319, 6114, 57, 719206527 [7.192e8], 6813, 16419199797069286018154632093538978307443519224016378057156716264399020428577 [1.641e76], 6575, 4202047189 [4.202e9], 1297, 3512, 3870, 5795, 6337, 5660, 5989, 6112, 8330, 13, 1877, 187, 264, 8048, 5850, 941, 6303, 1009, 7315, 8453, 5662, 3183, 8178, 94, 131073 [1.31e5], 517440284 [5.174e8], 2696, 191, 262, 21888159374819042681065900960079108671330771976540605510559380874944847741954 [2.188e76], 7937, 8489, 225, 6486, 2450, 1800, 866, 2354975218 [2.354e9], 249, 624, 8483, 68, 221, 4202047188 [4.202e9], 6080, 6132, 6549, 5499, 2696, 2355, 6253, 7510, 640807601 [6.408e8], 2806, 30047347705315867906698902152804724170386939875701103062026714790719945441280 [3.004e76], 3987, 12650941915662020058015862023665998998969191525479888727406889100124684769508 [1.265e76], 3426, 3116737219246058688143980610047278814429847607725717372614734046457215800098 [3.116e75], 1877, 21888075877798810139885396174900942254113179552665176677420557563313886988290 [2.188e76], 8084, 2825, 941, 17220337697351015657950521176323262483320249231368149235373741788599650842711 [1.722e76], 162, 2421, 464, 5204, 1046363170 [1.046e9], 29, 5339, 7225, 463067319384493612253 [4.63e20], 3, 5811, 7973, 1800, 6570, 4339, 6357, 1801, 445, 34, 2793, 1094, 230, 362, 624, 271, 2132, 3720, 29, 3548, 3488, 6934, 7878, 5849, 624, 8405, 548, 4019340710878518128881639338678 [4.019e30], 154, 68, 1299, 3232, 5824, 2034, 2163, 2439649223 [2.439e9], 21204235282094297871551205565717985242031228012903033270457635305745314480128 [2.12e76], 3871, 190, 7571, 2367]]] testFuzz_evalDataGen_matches(uint256,uint256,uint256[]) (runs: 123, μ: 2779926, ~: 1962863)

Encountered a total of 1 failing tests, 112 tests succeeded
error: Recipe `sol-test` failed on line 90 with exit code 1
```

### PR DESCRIPTION
**Pull request by alxiong**
Part of  #1739 
Closes #1797

### This PR:

- Point to the latest release of `jf-plonk` which modify the `PlonkTranscript` logic
- Remove endian reversal for field elements
- For G1Point, instead of using compressed form via `g1Serialize`, simply concatenate two base fields to the transcript byte array
- Update `_computeChallenges()` to reflect the same change

### Gas benchmark

```
@@ -1,2 +1,2 @@
-LightClient_newFinalizedState_Test:testCorrectUpdateBench() (gas: 594565)
-PlonkVerifier_verify_Test:test_verify_succeeds() (gas: 507806)
\ No newline at end of file
+LightClient_newFinalizedState_Test:testCorrectUpdateBench() (gas: 661721)
+PlonkVerifier_verify_Test:test_verify_succeeds() (gas: 496676)
```
I'm not 100% sure why the cost of PlonkVerifier goes down  but the LightClient cost goes up. 
(see if you have any idea @philippecamacho ?)

### CODE REVIEW

**Comment by alxiong** [comment]
Oh wow, this is such an unexpected behavior, somehow adding a `console.log` line pass all the test

```solidity
        transcript.transcript = abi.encodePacked(
            transcript.transcript, proof.wire3.x, proof.wire3.y, proof.wire4.x, proof.wire4.y
        );

        // UNCOMMENT THIS LINE, THEN ALL FAILED TEST PASSED
        // console2.logBytes(transcript.transcript);
        res.beta = transcript.getAndAppendChallenge();
        res.gamma = transcript.getAndAppendChallenge();
```

🤯 investigating why....

**Comment by alxiong** [comment]
Let me document a minimally reproducible bug (of my own code) that blocks me for days: 

```solidity
import "forge-std/Test.sol";
import { Transcript as T } from "../src/libraries/Transcript.sol";

library Transcript {
    struct TranscriptData {
        bytes transcript;
    }
    function getAndAppendChallenge(TranscriptData memory self) internal pure returns (uint256) {
        bytes memory transcript = self.transcript;
        uint256 ret = uint256(keccak256(transcript)) % BN254.R_MOD;

        assembly {
            let len := mload(transcript)
            let newLen := add(len, 32)
            let dataPtr := add(transcript, 0x20)

            mstore(transcript, newLen)
            mstore(add(dataPtr, len), ret)
        }
        return ret;
    }
}

contract Whatever is Test {
    // run with `forge test --mt whatever -vv`
    function test_whatever() external {
        T.TranscriptData memory transcript;
        transcript.transcript = abi.encodePacked(
            transcript.transcript,
            uint256(0x1234)
        );

        // HERE !! try to comment and uncomment
        console2.log("hi");

        uint256 chal = T.getAndAppendChallenge(transcript);
        console2.log("chal: %x", chal);
        console2.logBytes(transcript.transcript);
        revert(); // deliberately revert to print logs
    }
}
```
so comment out the `console2.log` before the `getAndAppendChallenge()`: you would get: 
```
  chal: 0x21909f356f36b5c053a1b997dd03e86a313c86db7ca957c0840832cca43b62c1
  0x0000000000000000000000000000000000000000000000000000000000001234
    0000000000000000000000000000000000000000000000000000000000000008
```
uncomment it, you would get the correct/expected transcript bytes:
```
  chal: 0x21909f356f36b5c053a1b997dd03e86a313c86db7ca957c0840832cca43b62c1
0x0000000000000000000000000000000000000000000000000000000000001234
21909f356f36b5c053a1b997dd03e86a313c86db7ca957c0840832cca43b62c1
```

File: contracts/src/libraries/Transcript.sol [hunk 1]
```diff
@@ -39,39 +37,24 @@ library Transcript {
...
-
-            // Copy 'self.transcript' to memory after 'self.state'
-            let dataOffset := add(data, 32) // Start right after 'self.state'
-            for { let i := 0 } lt(i, bLength) { i := add(i, 0x20) } {
-                mstore(add(dataOffset, i), mload(add(add(b, i), 0x20)))
-            }
-
-            // Compute the keccak256 hash of the data
-            hash := keccak256(data, add(32, bLength))
+            let len := mload(transcript)
```
Comment (philippecamacho) [inline contracts/src/libraries/Transcript.sol:48]: Suggestion: add a comment explaining what is being done in the assembly code, e,g.:

`// self.transcript = abi.encodePacked(self.transcript,ret);`

**Comment by philippecamacho** [comment]
Re benchmarks:

That is weird indeed. Commenting the [line](https://github.com/EspressoSystems/espresso-sequencer/blob/77a13cee3c16e226cbf7229d5fb7d3959e0c82b4/contracts/src/LightClient.sol#L275) that verifies the proof gives around 160k gas. So it looks like for some reason the proof verification in the light client contract spends more gas than in the Plonk verifier test...

```
Ran 1 test for contracts/test/LightClientBenchmark.t.sol:LightClient_newFinalizedState_Test
[PASS] testCorrectUpdateBench() (gas: 165444)
Suite result: ok. 1 passed; 0 failed; 0 skipped; finished in 14.29s (14.26s CPU time)
```

**Comment by philippecamacho** [comment]
> Re benchmarks:
> 
> When I comment these [three lines](https://github.com/EspressoSystems/espresso-sequencer/blob/cp-i02/contracts/test/mocks/LightClientMock.sol#L49-L51) the gas consumption is around 170k.
> 
> When I do the same in [b3ec621](https://github.com/EspressoSystems/espresso-sequencer/commit/b3ec621ceeb04f29c8d767f810c69d178f9cebd3) I get 90k. So there is around a 80k gas leakage in this [part](https://github.com/EspressoSystems/espresso-sequencer/blob/cp-i02/contracts/test/mocks/LightClientMock.sol#L36-L47) of the code. Weird.

@alxiong : Ticket created: https://github.com/EspressoSystems/espresso-sequencer/issues/1816 so that we can come back to it later.

File: contracts/src/libraries/Transcript.sol (hunk 1 above)
Comment (alxiong) [reply contracts/src/libraries/Transcript.sol:48]: will do, btw, if we replace with your line, the gas will jump up to:
```
LightClient_newFinalizedState_Test:testCorrectUpdateBench() (gas: 696716)
PlonkVerifier_verify_Test:test_verify_succeeds() (gas: 531948)
```

so our assembly did save gas!

Comment (philippecamacho) [reply contracts/src/libraries/Transcript.sol:48]: Definitely worth the effort!

Comment (alxiong) [reply contracts/src/libraries/Transcript.sol:52]: done in [5fba575](https://github.com/EspressoSystems/espresso-sequencer/pull/1801/commits/5fba575b75e47783b5c12d294169ed96dfe2787c)

**Pull request merged**